File: app/src/main/java/ru/visualmath/android/api/model/SlideInfo.java
package ru.visualmath.android.api.model;

import com.google.gson.annotations.SerializedName;

public class SlideInfo {

    @SerializedName("type")
    private SlideType type;

    public SlideType getType() {
        return type;
    }

    public enum SlideType {
        @SerializedName("slide")
        MODULE,

        @SerializedName("question")
        QUESTION,

        @SerializedName("questionBlock")
        QUESTION_BLOCK
    }
}

File: app/src/main/java/ru/visualmath/android/api/model/SlideInfo.kt
package ru.visualmath.android.api.model

import com.google.gson.annotations.SerializedName

data class SlideInfo(

        @SerializedName("type")
        val type: SlideType
) {

    enum class SlideType {
        @SerializedName("slide")
        MODULE,

        @SerializedName("question")
        QUESTION,

        @SerializedName("questionBlock")
        QUESTION_BLOCK,

        @SerializedName("visual")
        VISUAL
    }
}
